File: typescript/adapter/Banner.ts
export class Banner {
  protected text: string;

  constructor(text: string) {
    this.text = text;
  }

  showWithParen = (): string => {
    return `(${this.text})`;
  }

  showWithAster = (): string => {
    return `*${this.text}*`;
  }
}

File: go/adapter/banner/banner.go
package banner

import (
	"fmt"
)

type banner struct {
	text string
}

func New(text string) *banner {
	return &banner{text}
}

func (b *banner) ShowWithParen() string {
	return fmt.Sprintf("(%s)", b.text)
}

func (b *banner) ShowWithAster() string {
	return fmt.Sprintf("*%s*", b.text)
}
